news/2020-03-19-Micro_Profile_OpenTracing_Comes_To_WildFly.adoc

```diff
@@ -0,0 +1,113 @@
+  -p 14268:14268 \
+  -p 16686:16686 \
+  jaegertracing/all-in-one:1.16
+--
+
+Now, you can navigate to link:http://localhost:16686[http://localhost:16686] to access the Jaeger UI.
+
+== Configuring the OpenTracing subsystem
+
+You can use either UDP or TCP to send your log spans to Jaeger.  Note that those configurations are *exclusive*, so if you configure a TCP endpoint then Jaeger won't use the UDP configuration.
+
+=== Using UDP
+
+First we use the WildFly CLI tool to define an outbound socket binding towards the Jaeger tracer.
+[source,ruby]
+--
+[standalone@localhost:9990 /] /socket-binding-group=standard-sockets/remote-destination-outbound-socket-binding=jaeger:add(host=localhost, port=6831)
+{"outcome" => "success"}
+--
+
+Now we can define our MPOT tracer configuration:
+[source,ruby]
+--
+[standalone@localhost:9990 /] /subsystem=microprofile-opentracing-smallrye/jaeger-tracer=jaeger-demo:add(sampler-type=const, sampler-param=1, reporter-log-spans=true, sender-binding=jaeger)
+{"outcome" => "success"}
+--
+
+=== Using TCP
+
+If you want to use TCP instead of UDP you need to configure the *sender-endpoint* and set its value to *http://localhost:14268/api/traces*.
+
+[source,ruby]
+--
+[standalone@localhost:9990 /] /subsystem=microprofile-opentracing-smallrye/jaeger-tracer=jaeger-demo:write-attribute(name="sender-endpoint", value="http://localhost:14268/api/traces")
+{
+    "outcome" => "success",
+    "response-headers" => {
+        "operation-requires-reload" => true,
+        "process-state" => "reload-required"
+    }
+}
+[standalone@localhost:9990 /] reload
+--
+
+=== Setting the default tracer
+
+Let's define this new tracer as the default tracer to be used by WildFly:
+[source,ruby]
+--
+[standalone@localhost:9990 /] /subsystem=microprofile-opentracing-smallrye:write-attribute(name=default-tracer, value=jaeger-demo)
+{
+    "outcome" => "success",
+    "response-headers" => {
+        "operation-requires-reload" => true,
+        "process-state" => "reload-required"
+    }
+}
+[standalone@localhost:9990 /] reload
+--
+
+[NOTE]
+====
+Instead of relying on the default tracer attribute, you can use the *web.xml* deployment descriptor and set the init-parameter *smallrye.opentracing.tracer.configuration* to the name of the tracer to use.
+====
+
+== Deploying a demo application
+
+We provide a sample application at link:https://github.com/ehsavoie/opentracing-demo[https://github.com/ehsavoie/opentracing-demo].
+This application is a simple JAXRS endpoint producing XML or JSON and failing randomly.
+
+[source,bash]
+--
+git clone git@github.com:ehsavoie/opentracing-demo.git
+cd opentracing-demo
+mvn clean install
+cp target/opentracing-demo.war $WILDFLY_HOME/standalone/deployments
+$WILDFLY_HOME/bin/standalone.sh
+--
+
+Now, you can navigate to link:http://localhost:8080/opentracing-demo[http://localhost:8080/opentracing-demo] to access the deployed application. Generate a few traces by clicking on the links several times.
```

Comment:
Can you add a sentence something like this? "Note that the application deliberately throws exceptions randomly, so don't be surprised if some requests fail." Perhaps in italics?

I vaguely knew it would fail but it was still a bit surprising. And many people will likely not need carefully and won't notice the 'and failing randomly' above.

Also, https://github.com/ehsavoie/opentracing-demo/pull/2